src/main/java/marquez/db/DatasetVersionDao.java

```diff
@@ -92,6 +92,11 @@ default void insertWith(DatasetVersionRow row, List<DatasetFieldRow> fieldRows)
   @SqlQuery(SELECT + "WHERE uuid = :uuid")
   Optional<DatasetVersionRow> findBy(UUID uuid);
 
+  @SqlQuery(
+      SELECT
+          + "WHERE dataset_uuid = :datasetUuid AND created_at = (select max(created_at) from dataset_versions where dataset_uuid = :datasetUuid group by dataset_uuid) limit 1")
+  Optional<DatasetVersionRow> findLatestDatasetByUuid(UUID datasetUuid);
```

Comment:
the datasets table has a current_version_uuid so you can find the current version of the dataset with something along the lines of:
SELECT + " INNER JOIN datasets AS d ON d.uuid = dv.dataset_uuid AND d.current_version_uuid = dv.uuid WHERE dv.dataset_uuid = :uuid"

Reply:
Ah good call